### PR TITLE
Reenable cusparse SpMM on cuda 10.2

### PR DESCRIPTION
**Pull request by malfet**
This fixes feature regression introduced by #42412 which limited all the use of the API to CUDA-11.0+

### CODE REVIEW

**Review by xwang233 — APPROVED** [review]
Thank you so much for the update!

**Review by ngimel — APPROVED** [review]

File: aten/src/ATen/native/sparse/cuda/SparseCUDABlas.cu [hunk 1]
```diff
@@ -9,7 +9,17 @@
...
 #include <cusparse.h>
 
-#if defined(__CUDACC__) && CUSPARSE_VERSION >= 11000
+// LIMITATION (cusparseSpMM):
+// The generic APIs are available on all platforms on CUDA 11.0
+// For CUDA 10.1+ it is available for all platforms except Windows.
+// Using these APIs in any other systems will result in compile-time or run-time failures.
+// Their support will be extended in the next releases.
+
+#define IS_SPMM_AVAILABLE() (defined(__CUDACC__) &&                              \
```
Comment (peterjc123) [inline aten/src/ATen/native/sparse/cuda/SparseCUDABlas.cu:18]: Hey @malfet , looks like this macro caused the Windows failure. The weird thing is that if I replace `IS_SPMM_AVAILABLE()` with the content in it, it works.

Comment (peterjc123) [reply aten/src/ATen/native/sparse/cuda/SparseCUDABlas.cu:18]:
```cpp
#include <iostream>
#include <cusparse.h>

using namespace std;

#define IS_SPMM_AVAILABLE() (defined(__CUDACC__) &&                              \
                             (CUSPARSE_VERSION >= 11100 ||                       \
                              (!defined(_MSC_VER) && CUSPARSE_VERSION >= 10301)))

int main() {
    cout << CUSPARSE_VERSION << endl;
    cout << __CUDACC__ << endl;
    cout << _MSC_VER << endl;
    cout << (__CUDACC__ > 0 && CUSPARSE_VERSION >= 11000) << endl;
    cout << (_MSC_VER > 0 && CUSPARSE_VERSION >= 10301) << endl;
#if (defined(__CUDACC__) && (CUSPARSE_VERSION >= 11100 || (!defined(_MSC_VER) && CUSPARSE_VERSION >= 10301)))
    cout << "hey" << endl;
#endif
#if IS_SPMM_AVAILABLE()
    cout << "yes" << endl;
    return 0;
#else
    cout << "no" << endl;
    return 1;
#endif
}
gives the output:
```
11100
1
1926
1
1
hey
no
```
```

Comment (xwang233) [reply aten/src/ATen/native/sparse/cuda/SparseCUDABlas.cu:18]: I found something similar https://github.com/cisco/cjose/issues/63. 
A fix is at https://github.com/pytorch/pytorch/pull/42643.

Comment (peterjc123) [reply aten/src/ATen/native/sparse/cuda/SparseCUDABlas.cu:18]:
```cpp
#define __CUDACC__ 1

#if defined(__CUDACC__)
#error works
#endif

#if IS_CUDACC()
#error this one doesn't work
#endif
```

Comment (malfet) [reply aten/src/ATen/native/sparse/cuda/SparseCUDABlas.cu:18]: @peterjc123 yeah, I keep forgetting about this, but `defined()` is not a macro that returns a value nor is it a function.

**Pull request closed without merging**